samples/sample_definition_and_criteria.rst

```diff
@@ -39,36 +47,42 @@ Sample Criteria
         sample.kernel.cond_var:
           integration_platforms:
             - native_sim
-          tags: kernel condition_variables
+          tags:
+            - kernel
+            - condition_variables
           harness: console
           harness_config:
             type: one_line
             regex:
               - ".*Waited and joined with 3 threads"
 
-  **Guidelines for Samples:**
-    * Minimize the use of ``platform_allow`` and architecture filters.
-    * Do not mark the test as build only unless necessary.
-    * Any test case-specific configuration options are added using ``extra_args`` or
-      ``extra_configs`` options in the YAML file. The `prj.conf` file should have the in-common
-      configuration options.
-    * The sample itself can be evaluated using multiple configurations in the sample's YAML file.
-      For example, the sample is used to run with different logging modes using multiple scenarios
-      in ``samples/subsys/logging/syst``.
-    * Sample output can be validated leveraging the ``harness_config`` regex option,
-      wherever applicable.
-
-3. Samples should target multiple platforms and architectures.
-++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
-    of testing to the mentioned platforms and architectures.
-    Reference: :ref:`twister_script`
+  * Do not mark the test as build only. A sample should be able to build and
+    run on the documented platforms. Use the harness option to skip the
+    execution. Twister only attempts to flash and execute the build binary if
+    the harness is set to ztest or console.
+  * The default configuration for the sample should be in the ``prj.conf``
```

Comment:
```suggestion
  * The default configuration for the sample must be in the :file:`prj.conf`
```

Also, as a general comment: maybe all the occurences of SHOULD / MUST / DO NOT, etc. at the beginning of each bullet point should be put in capital letters?